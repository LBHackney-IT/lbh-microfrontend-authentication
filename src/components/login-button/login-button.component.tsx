import React from 'react';

import { LoginButtonProperties } from './login-button.types';

import './login-button.scss';

export const LoginButton = ({
    loginUrl,
}: LoginButtonProperties): JSX.Element => (
    <a href={loginUrl} className="govuk-button lbh-button">
        Sign in using Hackney.gov.uk
    </a>
);
