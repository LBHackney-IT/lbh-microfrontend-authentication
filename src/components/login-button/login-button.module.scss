@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/core/all";
@import "~lbh-frontend/lbh/objects/all";

.button {
  display: block;
  max-width: 300px;

  @include lbh-font;
  border-radius: 4px;
  @include lbh-rem(font-size, 19);
  font-weight: $lbh-font-weight-regular;
  @include lbh-rem(line-height, 38);

  @include lbh-rem(padding, 13, 22, 11);
  border: 1px solid lbh-colour("lbh-a01");

  color: lbh-colour("lbh-a01");
  background: lbh-colour("lbh-white");

  padding-left: 60px;
  margin: 0 auto;

  &:hover {
    color: lbh-colour("lbh-white");
    background: lbh-colour("lbh-primary-button");
  }

  &:before {
    display: block;
    width: 40px;
    height: 40px;
    content: " ";
    background: url("./google-logo.svg") no-repeat center center transparent;
    margin-left: -50px;
    float: left;
  }
}
