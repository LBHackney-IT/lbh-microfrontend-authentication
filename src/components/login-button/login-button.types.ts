export type LoginButtonProperties = {
    loginUrl: string;
};
