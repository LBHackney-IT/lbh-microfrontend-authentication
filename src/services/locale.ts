const locale = {
  unauthorisedToViewService: "You do not have permission to access this service.",
  contactToRequestPermission: "To request access please complete this form",
  signInUsingHackney: "Sign in using Hackney.gov.uk",
  requestAccess: "Request access",
};

export default locale;
