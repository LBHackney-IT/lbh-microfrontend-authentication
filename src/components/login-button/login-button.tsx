import React from 'react';

import { login, Button } from '@mtfh/common';
import { locale } from '../../services';
import './login-button.scss';

export const LoginButton = (): JSX.Element => (
    <Button onClick={() => login()}>{locale.signInUsingHackney}</Button>
);
