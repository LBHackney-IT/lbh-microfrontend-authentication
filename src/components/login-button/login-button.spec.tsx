import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { login } from '@mtfh/common';
import { LoginButton } from './login-button';
import { locale } from '../../services';

jest.mock('@mtfh/common', () => ({
    ...jest.requireActual('@mtfh/common'),
    login: jest.fn(),
}));

it('renders correctly', () => {
    const { container } = render(<LoginButton />);
    expect(container).toMatchSnapshot();
});

it('should call login', async () => {
    const { getByText } = render(<LoginButton />);
    const button = getByText(locale.signInUsingHackney);
    fireEvent.click(button);
    expect(login).toHaveBeenCalled();
});
