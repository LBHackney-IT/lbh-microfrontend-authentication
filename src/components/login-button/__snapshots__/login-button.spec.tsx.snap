// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    class="govuk-button lbh-button"
    type="button"
  >
    Sign in using Hackney.gov.uk
  </button>
</div>
`;
